packages/backend-deployer/src/cdk_error_mapper.ts

```diff
@@ -50,6 +50,34 @@ export class CdkErrorMapper {
       errorName: 'SyntaxError',
       classification: 'ERROR',
     },
+    {
+      errorRegex: /\[ERR_MODULE_NOT_FOUND\]:(.*)\n/,
+      humanReadableErrorMessage: 'Cannot find module',
+      resolutionMessage:
+        'Check your backend definition in the `amplify` folder for missing file or package imports. Try running npm or yarn install.',
+      errorName: 'ModuleNotFoundError',
+      classification: 'ERROR',
+    },
+    {
+      // Truncate the cdk error message's second line (Invoke the CLI in sequence, or use '--output' to synth into different directories.)
+      errorRegex:
+        /Another CLI (.*) is currently(.*)\. |Other CLIs (.*) are currently reading from(.*)\. /,
+      humanReadableErrorMessage: 'Multiple sandbox instances detected.',
+      resolutionMessage:
+        'Make sure only one instance of sandbox is running for this project',
+      errorName: 'MultipleSandboxInstancesError',
+      classification: 'ERROR',
+    },
+    {
+      // Also extracts the first line in the stack where the error happened
+      errorRegex: /\[esbuild Error\]: ((?:.|\n)*?at .*)/,
+      humanReadableErrorMessage:
+        'Unable to build the Amplify backend definition.',
+      resolutionMessage:
+        'Check your backend definition in the `amplify` folder for syntax and type errors.',
+      errorName: 'ESBuildError',
+      classification: 'ERROR',
```

Comment:
Should this be pointing customer to their functions? Would ESBuild have any other "root of compilation" than functions?

Reply:
EsBuild "compiles" the entire project before building/bundling the functions. E.g.  typescript errors in the `backend.ts` were being caught by the ESBuild if TS is turned off.